test/integration/integration_test.go

```diff
@@ -72,3 +74,31 @@ func parseRunCommand(t *testing.T, out string) (string, string) {
 	stdError := strings.TrimSpace(split[2])
 	return stdOutput, stdError
 }
+
+func TestConfigImport(t *testing.T) {
+	if os.Getenv("AZURE_SUBSCRIPTION_ID") == "" || os.Getenv("AZURE_RESOURCE_GROUP") == "" || os.Getenv("AZURE_CLUSTER_NAME") == "" {
+		t.Skip("Skipping test because AZURE_SUBSCRIPTION_ID, AZURE_RESOURCE_GROUP, and AZURE_CLUSTER_NAME environment variables must be set")
+	}
+
+	configPath := filepath.Join(config.Dir(), "config.yaml")
+	defer os.Remove(configPath)
+
+	runCommand(t, os.Getenv("KUBECTL_AKS"), "config", "import")
+	k8sConfigFile, err := os.ReadFile(configPath)
+	require.Nil(t, err, "reading config file: %v", err)
+	require.NotEmpty(t, k8sConfigFile, "config file is empty")
+
+	runCommand(t, os.Getenv("KUBECTL_AKS"), "config", "unset-all")
+	_, err = os.ReadFile(configPath)
+	require.NotNil(t, err, "reading config file: %v", err)
+
+	// set KUBECONFIG to a non-existent file to make sure it doesn't get used
+	os.Setenv("KUBECONFIG", "foo")
+	_, stderr := runCommand(t, os.Getenv("KUBECTL_AKS"), "config", "import", "-s", os.Getenv("AZURE_SUBSCRIPTION_ID"), "-g",
+		os.Getenv("AZURE_RESOURCE_GROUP"), "-c", os.Getenv("AZURE_CLUSTER_NAME"))
```

Comment:
This is the part I find a little bit strange. Even if I pass the parameters to get it through the Azure API, the command will anyway try to use the k8s mechanism and wait until it times out before trying the Azure API mechanism.

IMO, we should allow users using directly the Azure API mechanism.